src/csvinputfileconn.h

```diff
@@ -277,44 +292,33 @@ namespace dd
 	}
     }
 
-    void shuffle_data(const APIData &ad)
+    void shuffle_data(std::vector<CSVline> &csvdata,
+                      const bool forbid_shuffle = false)
     {
-      if (ad.has("shuffle") && ad.get("shuffle").get<bool>())
-	{
-	  std::mt19937 g;
-	  if (ad.has("seed") && ad.get("seed").get<int>() >= 0)
-	    {
-	      g = std::mt19937(ad.get("seed").get<int>());
-	    }
-	  else
-	    {
-	      std::random_device rd;
-	      g = std::mt19937(rd());
-	    }
-	  std::shuffle(_csvdata.begin(),_csvdata.end(),g);
-	}
+      if (_shuffle && !forbid_shuffle)
```

Comment:
I don't like it too much, let's control the call to `shuffle_data()` instead, if possible.